doc/object_lifetime_management.md

```diff
@@ -0,0 +1,80 @@
+# Object lifetime management
+
+As the methods and classes within the node-addon-api are used,
+handles to objects in the heap for the underlying
+VM may be created. A handle may be created for any new node-addon-api
+Value(and subclasses) created or returned. These handles must hold the
+objects 'live' until they are no longer required by the native code,
+otherwise the objects could be collected before the native code was
+finished using them.
```

Comment:
otherwise the objects could be collected by the garbage collector before the native code was finished using them.